core/lib/rom/plugins/command/timestamps.rb

```diff
@@ -70,10 +70,9 @@ def datestamp_columns
           def set_timestamps(tuples, *)
             timestamps = build_timestamps
 
-            case tuples
-            when Hash
+            if tuples.respond_to?(:merge)
```

Comment:
@cflipse Just curious, would this snippet be clearer? 
```ruby
timestamps.merge(tuples) if tuples.respond_to?(:merge)
tuples.map { |t| timestamps.merge(t) } if tuples.respond_to?(:map)
```

Reply:
responding to one or the other method are not mutually exclusive.

Reply:
there's a method called `Command#with_input_tuples` which yields hashes, IIRC it should be available in this context

Reply:
If I remember right, `Command#with_input_tuples` is a private method defined inside the create & update commands in rom-sql.

Reply:
Bummer. I’d say it should be moved to rom-core then :)

Reply:
👍 plus it has the added benefit of removing the duplication between the rom-sql create and update commands.

Reply:
Just did a trial run of yanking that over.

One downside:  The method as-is transforms a single-hash tuple into an array of tuples ... which breaks a few specs, because they (rightly) assume the airity of the tuples remains consistant

Reply:
@cflipse gah right, well then we need something that does `with_input_tuples` + returns either one or many results :/ Remember when I said I regret adding `result` option to commands? :D This is precisely how it complicates commands.

Reply:
it's worse than that, actually:

the mapping / result needs to agree with it's argument.  `update`  command,  for example can generally be considered to return `:many` but may only be called with a single input tuple.

I can get specs to pass by checking on the hashiness of the input tuple, but I suspect this might end up causing problems with pipelines ... though maybe those are supposed to be treated as more array-like anyway?

Reply:
Aaaand, I can't use the `with_input_tuples` name because rom-sql has already internalized the idea that it always returns an array, and so it breaks multi-insert.  

`map_input_tuples` it is.